src/test/java/ch/heigvd/pro/b04/questions/ServerQuestionTest.java

```diff
@@ -134,14 +133,52 @@ public void testByIdEndpointsReturnTheRightQuestions()
     when(pollRepo.findById(pollTemp.getIdPoll())).thenReturn(Optional.of(pollTemp));
     when(modoRepo.findByToken("t1")).thenReturn(Optional.of(aloy));
     when(participantRepository.findByToken("t1")).thenReturn(Optional.empty());
-    when(repo.findById(new ServerQuestionIdentifier(1))).thenReturn(Optional.of(c1));
+    when(questionRepository.findById(new ServerQuestionIdentifier(1))).thenReturn(Optional.of(c1));
 
     assertEquals(c1, cc.byId("t1", pollTemp.getIdPoll(), new ServerQuestionIdentifier(1)));
     //with wrong idQuestion
     assertThrows(ResourceNotFoundException.class,
         () -> cc.byId("t1", pollTemp.getIdPoll(), new ServerQuestionIdentifier(3)));
   }
 
+  @Test
+  public void testParticipantCanAlsoAccessQuestions()
+      throws WrongCredentialsException, ResourceNotFoundException, PollNotExistingException {
+    ServerQuestion c1 = ServerQuestion.builder()
+        .id(1).title("Do you dream of Scorchers ?").build();
+    ServerQuestion c2 = ServerQuestion.builder()
+        .id(2).title("Do you love the Frostclaws ?").build();
+
+    ServerPoll pollTemp = ServerPoll.builder()
+        .idPoll(ServerPollIdentifier.builder().idPoll(123).build())
+        .pollServerQuestions(Set.of(c1, c2)).build();
+
+    Moderator ikrie = Moderator.builder()
+        .idModerator(2)
+        .username("ikrie")
+        .pollSet(Set.of(pollTemp))
+        .secret("banuk").build();
+
+    ServerSession s2 = ServerSession.builder()
+        .idSession(SessionIdentifier.builder()
+            .idSession(1).idxPoll(pollTemp).idxModerator(ikrie).build())
+        .state(SessionState.OPEN).build();
+
+    Participant aloy = Participant.builder()
+        .idParticipant(ParticipantIdentifier.builder()
+            .idParticipant(1).idxServerSession(s2).build())
+        .token("t1").build();
+
+    when(pollRepo.findById(pollTemp.getIdPoll())).thenReturn(Optional.of(pollTemp));
+    lenient().when(modoRepo.findByToken("t2")).thenReturn(Optional.of(ikrie));
+    when(participantRepository.findByToken("t1")).thenReturn(Optional.of(aloy));
+    when(questionRepository.findById(new ServerQuestionIdentifier(1))).thenReturn(Optional.of(c1));
```

Comment:
Why is the identifier created with only a discriminator, and no poll / moderator keys ?